### DISCUSSION TITLE
Way to use drivebase.straight with heading angles

### DISCUSSION THREAD

**Discussion by Mathi256** [pybricks/support#1719] [Ideas]
Hello, I am part of a WRO team and use the drive base functions frequently. These functions are very useful to us since it allows us to make very precise movements. For the most part, I think the list of functions are very complete, except for one thing. **I think there should be a way to make the robot move straight some amount of distance, but while following a given heading angle.** I know there is a way to activate gyro in the drivebase, but this one only follows the angle that it started in. So, for example, if it was facing 78 degrees it would move in that direction. What I want is something that if it started at any angle (78 degrees for example) it would first face the angle I defined (90 for example) and then move forward. This would help us at WRO since if we told the robot to move forward and it was misaligned, it would auto-align and then move forward.

**Comment by laurensvalk** [comment]
>  What I want is something that if it started at any angle (78 degrees for example) it would
> - first face the angle I defined (90 for example)
> - and then move forward. 

You could do each of those in separate steps, right?

The trick to the first step is to `turn` by the _remaining_ angle to your target:

```python
robot.turn(90 - robot.angle())
```

**Reply by Mathi256** [reply]
Yes, I think that works. My only fear is that when it turns it turns like 1 degree more, so when it moves forward it wouldn’t move straight. My suggestion is to make a drive straight but while following a given angle so it doesn’t drive 1 degree off.

**Reply by laurensvalk** [reply]
When using the default `Stop.HOLD`, the intermediate targets are preserved, so making a whole new method would achieve the same as these two separately.

I think other aspects such as the natural drift or hub misalignment are more likely to induce a minimal 1 degree error :)

**Reply by Mathi256** [reply]
Thank you for your idea! I'm going to test using stop.hold before moving straight to see if this works.
